scripts/npm/create-component.js

```diff
@@ -0,0 +1,156 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2018 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* @flow */
+
+/* eslint-disable no-console */
+
+const { execSync } = require('child_process');
+const prompt = require('prompt');
+const fs = require('fs');
+const colors = require('colors');
+
+const STORYBOOK_CONFIG_SPLIT_POINT_1 = 'configure(() => {';
+const STORYBOOK_CONFIG_SPLIT_POINT_2 = '}, module);';
+
+const schema = {
+  properties: {
+    platform: {
+      description: 'native or web?',
+      default: 'native',
+      pattern: /^(native|web)$/,
+      message: 'Enter "native" or "web"',
+    },
+    name: {
+      description: "Enter the component name, e.g. 'banner-alert'",
+      pattern: /^[a-z-]+$/,
+      message: 'Use snake case, e.g. "banner-alert".',
+      required: true,
+    },
+  },
+};
+
+const snakeCaseToPascalCase = input => {
+  const re = /(-\w)/g;
+  const convert = matches => matches[1].toUpperCase();
+  const camelCase = input.replace(re, convert);
+  const pascalCase = camelCase[0].toUpperCase() + camelCase.slice(1);
+  return pascalCase;
+};
+
+const createComponent = (err, { platform, name }) => {
+  if (err) {
+    console.error(err);
+    return;
+  }
+
+  let boilerplateComponentPath = `packages/bpk-component-boilerplate`;
+  let newComponentPath = `packages/bpk-component-${name}`;
+  let storybookConfigFile = `.storybook/config.js`;
+  let storybookImport = `require('./../${newComponentPath}/stories');`;
+
+  if (platform === 'native') {
+    boilerplateComponentPath = `native/packages/react-native-bpk-component-boilerplate`;
+    newComponentPath = `native/packages/react-native-bpk-component-${name}`;
+    storybookConfigFile = `native/storybook/storybook.js`;
+    storybookImport = `require('../packages/react-native-bpk-component-${name}/stories');`;
+  }
+
+  const pascalCaseName = snakeCaseToPascalCase(name);
+
+  const replaceInFiles = (oldWord, newWord) =>
+    `LC_CTYPE=C; LANG=C; find ${newComponentPath} -type f -exec sed -i "" "s/${oldWord}/${newWord}/g" {} +`;
+
+  const renameFiles = (oldWord, newWord) => {
```

Comment:
I would argue that a Stream can do a much better job here that a bunch of commands, the problem with commands is they can behave differently on different machines, a stream will make the code neater and consistent mostly everywhere.
I might be too opinionated here, see what the other thinks

Reply:
I'm not well-versed in Streams, have you got a link or anything I can look at?

Reply:
We use them in the svg transforming https://github.com/Skyscanner/backpack/blob/b88e0116f2bcfde3e380800cbad69fbb15ba7733/packages/bpk-svgs/tasks/svg2datauri.js#L72

I'm more than happy to pair with you if we want to go down the Streams route